src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/WindowBackdrop.cs

```diff
@@ -0,0 +1,404 @@
+
+            if (windowHandle == IntPtr.Zero)
+            {
+                return;
+            }
+
+            ApplyBackdrop(windowHandle, backdropType);
+        };
+
+        return true;
+    }
+
+    /// <summary>
+    /// Applies backdrop effect to the selected handle.
+    /// </summary>
+    /// <param name="hWnd">Window handle.</param>
+    /// <returns><see langword="true"/> if the operation was successfull, otherwise <see langword="false"/>.</returns>
+    internal static bool ApplyBackdrop(IntPtr hWnd, WindowBackdropType backdropType)
+    {
+        if (hWnd == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        if (!NativeMethods.IsWindow(hWnd))
+        {
+            return false;
+        }
+
+        if (ThemeColorization.IsThemeDark())
+        {
+            _ = UnsafeNativeMethodsWindow.ApplyWindowDarkMode(hWnd);
+        }
+        else
+        {
+            _ = UnsafeNativeMethodsWindow.RemoveWindowDarkMode(hWnd);
+        }
+
+        EnableGlassFrame(hWnd, backdropType);
+
+        // 22H1
+        if (!Utility.IsOSWindows11Insider1OrNewer)
+        {
+            if (backdropType != WindowBackdropType.None)
+            {
+                return ApplyLegacyMicaBackdrop(hWnd);
+            }
+
+            return false;
+        }
+
+        switch (backdropType)
+        {
+            case WindowBackdropType.Auto:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_AUTO);
+
+            case WindowBackdropType.Mica:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_MAINWINDOW);
+
+            case WindowBackdropType.Acrylic:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_TRANSIENTWINDOW);
+
+            case WindowBackdropType.Tabbed:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_TABBEDWINDOW);
+        }
+
+        return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_DISABLE);
+    }
+
+    /// <summary>
+    /// Tries to remove backdrop effects if they have been applied to the <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window from which the effect should be removed.</param>
+    internal static bool RemoveBackdrop(System.Windows.Window window)
+    {
+        if (window is null)
+        {
+            return false;
+        }
+
+        IntPtr windowHandle = new WindowInteropHelper(window).Handle;
+
+        return RemoveBackdrop(windowHandle);
+    }
+
+    /// <summary>
+    /// Tries to remove all effects if they have been applied to the <c>hWnd</c>.
+    /// </summary>
+    /// <param name="hWnd">Pointer to the window handle.</param>
+    internal static bool RemoveBackdrop(IntPtr hWnd)
+    {
+        if (hWnd == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        _ = RestoreContentBackground(hWnd);
+
+        if (hWnd == IntPtr.Zero)
```

Comment:
Is this redundant judgment? We check the `hWnd == IntPtr.Zero`  in line 155

Reply:
Yes this is, I will go ahead and remove this.